### PR TITLE
Implement --output-profile

### PR DESCRIPTION
**Pull request by mizdra**
I understand that this is my own idea and has not yet been endorsed by anyone. However, I think this feature is worth implementing in `backstop-retry-failed-scenarios` and decided to send a PR. If you do not need this PR, please ignore it :)

## Summary
- Enable backstop-retry-failed-scenarios to generate profiles so that backstop performance can be visualized

## Motivation
- Often backstop is very heavy and consumes a large portion of CI's processing time.
- Measuring the execution time of backstop is very important to improve the processing time of CI.
- In other words, it would be useful to have `backstop-retry-failed-scenarios` to measure the execution time of backstops.
- The purpose of this PR is to output a detailed report of how long some of the steps performed inside `backstop-retry-failed-scenarios` took.

## TODO
- [x] Implement `--output-profile`
- [x] Add test
- [ ] Add documentation
  - I'll write after the PR is accepted. 
- (If you need something else to do)

### CODE REVIEW

File: src/test/Runner.test.ts [hunk 1]
```diff
@@ -57,6 +60,52 @@ describe('Runner', () => {
...
+        config: 'backstop.json',
+        referenceCommand: 'not_existing_command',
+        command: 'not_existing_command',
+        outputProfile,
+        rootDir: resolve('backstop/failed'),
+      });
+      await runner.run();
+      expect(JSON.parse(fs.readFileSync(outputProfile).toString())).toEqual([
+        {
+          label: 'backstop.run',
```
Comment (mizdra) [inline src/test/Runner.test.ts:78]: I added a `backstop.` prefix for future expansion, but it may be a bit redundant.

File: src/cli.ts [hunk 2]
```diff
@@ -37,6 +37,12 @@ const optionDefinitions = [
     description:
       'Command to create reference before testing. Default: null (Do not create reference before test).',
   },
+  {
+    name: 'output-profile',
+    type: String,
+    defaultValue: null,
+    description: 'Measure the execution time and output the report as JSON.',
```
Comment (hitode909) [inline src/cli.ts:44]: I want to describe that this option is a path to output file, not boolean, or something.

For example:
```suggestion
    description: 'Path to profiler output file. If present, measure the execution time and output the report as JSON.',
```

File: src/test/Runner.test.ts (hunk 1 above)
Comment (hitode909) [reply src/test/Runner.test.ts:78]: I think `run` is enough, because this software is a backstop runner.

**Review by hitode909 — COMMENTED** [review]
Thank you for contribution, I wrote some comment.

File: src/cli.ts (hunk 2 above)
Comment (mizdra) [reply src/cli.ts:44]: Oh, I think it was originally implemented as a bool flag and has been retained. I'll fix it.

**Comment by mizdra** [comment]
Thank you for comment! I fixed a the problems. Can you please re-review?

**Review by hitode909 — APPROVED** [review]
LGTM!
Would you like to write some description around here?
- https://github.com/hitode909/backstop-retry-failed-scenarios/blame/master/README.md#L68

**Comment by mizdra** [comment]
I forgot it! I've added the documentation now.

**Comment by hitode909** [comment]
Thank you! I'll merge and release it.
